src/meta/app/src/share/share.rs

```diff
@@ -611,3 +628,38 @@ pub struct ShareInfo {
     pub name_ident: ShareNameIdent,
     pub meta: ShareMeta,
 }
+
+#[derive(serde::Serialize, serde::Deserialize, Clone, Debug, Default, Eq, PartialEq)]
+pub struct ShareTableSpec {
+    pub name: String,
+    pub id: u64,
+    pub location: String,
+    pub version: u64,
+    pub presigned_url_timeout: String,
+}
+
+impl ShareTableSpec {
+    pub fn new(name: &str, id: u64, location: String) -> Self {
+        ShareTableSpec {
+            name: name.to_owned(),
+            id,
+            location,
+            version: 1,
+            presigned_url_timeout: "120s".to_string(),
+        }
+    }
+}
+
+#[derive(serde::Serialize, serde::Deserialize, Clone, Debug, Default, Eq, PartialEq)]
+pub struct ShareDatabaseSpec {
+    pub name: String,
+    pub id: u64,
+}
+
+#[derive(serde::Serialize, serde::Deserialize, Clone, Debug, Default, Eq, PartialEq)]
+pub struct ShareSpec {
+    pub name: String,
+    pub database: Option<ShareDatabaseSpec>,
+    pub tables: Vec<ShareTableSpec>,
+    pub tenants: Vec<String>,
```

Comment:
besides the tenants list, a share may also be granted to ALL, which allows everyone to access this share.

how about adds a `pub global: bool` field?